code/API_definitions/device-status.yaml

```diff
@@ -120,38 +310,43 @@ components:
     ActiveRoaming:
       description: Roaming status. True, if it is roaming
       type: boolean
-    UeId:
+    Device:
       description: User equipment identifier
       type: object
       properties:
-        externalId:
-          $ref: '#/components/schemas/ExternalId'
-        msisdn:
-          $ref: '#/components/schemas/MSISDN'
-        ipv4Addr:
-          $ref: '#/components/schemas/Ipv4Addr'
-        ipv6Addr:
-          $ref: '#/components/schemas/Ipv6Addr'
+        networkAccessIdentifier:
+          $ref: '#/components/schemas/NetworkAccessIdentifier'
+        phoneNumber:
+          $ref: '#/components/schemas/PhoneNumber'
+        ipv4Address:
+          $ref: '#/components/schemas/Ipv4Address'
+        ipv6Address:
+          $ref: '#/components/schemas/Ipv6Address'
       minProperties: 1
-    ExternalId:
+    NetworkAccessIdentifier:
       type: string
       example: '123456789@domain.com'
-    MSISDN:
+    PhoneNumber:
       type: string
       pattern: '^\+?[0-9]{5,15}$'
       example: '123456789'
       description: Subscriber number in E.164 format (starting with country code). Optionally prefixed with '+'.
-    Ipv4Addr:
-      type: string
-      format: ipv4
-      pattern: '^([01]?\d\d?|2[0-4]\d|25[0-5])(?:\.(?:[01]?\d\d?|2[0-4]\d|25[0-5])){3}(\/([0-9]|[1-2][0-9]|3[0-2]))?$'
-      example: '192.168.0.1/24'
-      description: |
-        IPv4 address may be specified in form <address/mask> as:
-          - address - an IPv4 number in dotted-quad form 1.2.3.4. Only this exact IP number will match the flow control rule.
-          - address/mask - an IP number as above with a mask width of the form 1.2.3.4/24.
-            In this case, all IP numbers from 1.2.3.0 to 1.2.3.255 will match. The bit width MUST be valid for the IP version.
-    Ipv6Addr:
+    Ipv4Address:
```

Comment:
Allowing `privateAddress` as a device identifier makes it easier for the device itself (or an app on the device) to make the API call. But for this API, I don't see why the device itself would need to use an API call to the network to determine its roaming (or connectivity) status. So I don't think it is needed here to support that scenario.

Having said that, an API consumer other than the device itself may know the private IP address, and so this would still be a valid way of identifying the device. In fact, it is easier for the network to identify the device by `privateAddress` + `publicAddress` compared to `publicAddress` + `publicPort`, so we should encourage use of `privateAddress` if that is known.

So my preference is to follow QoD and Device Location when it comes to identifying the device by `privateAddress`, even if the justification is not so strong.

Reply:
updated